packages/edit-site/src/components/header/index.js

```diff
@@ -89,7 +91,10 @@ export default function Header( {
 		<div className="edit-site-header">
 			<div className="edit-site-header_start">
 				<MainDashboardButton.Slot>
-					<FullscreenModeClose />
```

Comment:
This button is exported from the package (experimental for now) and provides a way for 3rd plugins to customize some of its parts while retaining functionality ([see docs here](https://github.com/WordPress/gutenberg/blob/65258a077e4899ee4933c79077108f56f8b5ce6d/docs/designers-developers/developers/slotfills/main-dashboard-button.md)). Given that it will no longer represent a close action, I think that we should rename the existing `FullscreenModeClose` to `NavigationButton ` or `DashboardButton` and place our code changes there.

Reply:
We should also figure out how to handle the FSE Navigation when not in fullscreen mode, where we wouldn't have the WP/Site Icon back button anymore.

Reply:
Good point @Copons! I don't consider it a blocker for this PR though, as it's something that we can figure out later.